### PR TITLE
support add instructions at the end of tasks.test in the build.gradle.kts

### LINKED ISSUE #9813: Implement the option to add new instructions at the end of `tasks.test` in the `build.gradle.kts`
### I suggest changing the existing GradleConfiguration API.

from:

```java
.gradleConfigurations()
  .configuration()
.and()
```

to:

```java
.configurations()
  .addConfigurarion("")
  .addTasksTestInstruction("")
.and()
```

### Here is how the changes will affect the existing code:

[At LocalProfileModuleFactory](https://github.com/jhipster/jhipster-lite/blob/main/src/main/java/tech/jhipster/lite/generator/server/springboot/localeprofile/domain/LocalProfileModuleFactory.java#L27-L43)  from:
```java
.gradleConfigurations()
  .configuration(
    """
    tasks.build {
      dependsOn("processResources")
    }

    tasks.processResources {
      filesMatching("**/*.yml") {
        filter { it.replace("@spring.profiles.active@", springProfilesActive) }
      }
      filesMatching("**/*.properties") {
        filter { it.replace("@spring.profiles.active@", springProfilesActive) }
      }
    }
    """
  )
```

to:

```java
.gradleConfigurations()
  .addConfigurarion(
    """
    tasks.build {
      dependsOn("processResources")
    }

    tasks.processResources {
      filesMatching("**/*.yml") {
        filter { it.replace("@spring.profiles.active@", springProfilesActive) }
      }
      filesMatching("**/*.properties") {
        filter { it.replace("@spring.profiles.active@", springProfilesActive) }
      }
    }
    """
  )
```

[At JacocoModuleFactory](https://github.com/jhipster/jhipster-lite/blob/main/src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactory.java#L79-L82) from:
```java
 .optionalReplacements()
  .in(path("build.gradle.kts"))
  .add(EXISTING_BUILD_GRADLE_TASK_TEST_NEEDLE, ADD_JACOCO_BUILD_GRADLE_TASK_TEST)
  .and()
.and()
```

to:
```java
.configurations()
  .addTasksTestInstruction("""finalizedBy("jacocoTestReport")""")
.and()
```
 
The goal of this issue is to [make this possible](https://github.com/renanfranca/sample-frontend-gradle/pull/2/files#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06L123-L124). Implement the option to add  instructions at the end of `tasks.test` in the `build.gradle.kts`

### PR DESCRIPTION
**Pull request by renanfranca**
Fix #9813

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/module/domain/javabuild/command/AddGradleConfiguration.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 public record AddGradleConfiguration(String configuration) implements JavaBuildCommand {
   public AddGradleConfiguration {
-    Assert.notNull("configuration", configuration);
+    Assert.notNull("instruction", configuration);
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/module/domain/javabuild/command/AddGradleConfiguration.java:7]:
```suggestion
    Assert.notNull("configuration", configuration);
```

File: src/main/java/tech/jhipster/lite/module/domain/javabuild/command/AddTasksTestInstruction.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package tech.jhipster.lite.module.domain.javabuild.command;
+
+import tech.jhipster.lite.shared.error.domain.Assert;
+
+public record AddTasksTestInstruction(String instruction) implements JavaBuildCommand {
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/module/domain/javabuild/command/AddTasksTestInstruction.java:5]:
```suggestion
public record AddGradleTasksTestInstruction(String instruction) implements JavaBuildCommand {
```

**Review by murdos — APPROVED** [review]
Thanks @renanfranca ! 🙂

**Comment by renanfranca** [comment]
> Thanks @renanfranca ! 🙂

You are welcome! 😃

**Pull request merged**